### PR TITLE
Token Caching

### PR DESCRIPTION
**Pull request by raakella1**

### CODE REVIEW

File: src/auth_manager/auth_manager.cpp [hunk 1]
```diff
@@ -2,14 +2,59 @@
...
+    const std::string error_;
+};
+
+AuthManager::AuthManager() :
+        m_cached_tokens(SECURITY_DYNAMIC_CONFIG(auth_manager->auth_token_cache_size)),
+        m_cached_keys(SECURITY_DYNAMIC_CONFIG(auth_manager->auth_key_cache_size)) {}
+
 AuthVerifyStatus AuthManager::verify(const std::string& token, std::string& msg) const {
+    // if we have it in cache, just use it to make the decision
+    auto const token_hash = md5_sum(token);
```
Comment (szmyd) [inline src/auth_manager/auth_manager.cpp:41]: Do we need a secure hash or std::hash would suffice?

File: src/auth_manager/security_config.fbs [hunk 2]
```diff
@@ -33,6 +33,10 @@ table AuthManager {
 
     // ssl verification for the signing key download url
     verify: bool = true;
+
+    // LRUCache sizes
+    auth_token_cache_size: uint32 = 2000;
+    auth_key_cache_size: uint32 = 100;
```
Comment (szmyd) [inline src/auth_manager/security_config.fbs:39]: Are these correct?

**Review by szmyd — COMMENTED** [review]
Questions

File: src/auth_manager/auth_manager.cpp (hunk 1 above)
Comment (raakella1) [reply src/auth_manager/auth_manager.cpp:41]: We can use std::hash

**Review by szmyd — APPROVED** [review]
lgtm

File: src/auth_manager/auth_manager.cpp (hunk 1 above)
Comment (raakella1) [reply src/auth_manager/auth_manager.cpp:41]: edit: std::hash will not work, we need a secure hash

**Pull request merged**